hybrasyl/Messaging/Effects.cs

```diff
@@ -0,0 +1,155 @@
+        public new static string Command = "effect";
+        public new static string ArgumentText = "<byte effect>";
+        public new static string HelpText = "Displays the specified effect (animation).";
+        public new static bool Privileged = false;
+
+        public new ChatCommandResult Run(User user, params string[] args)
+        {
+            short speed = 20;
+            if (byte.TryParse(args[0], out byte effect))
+            {
+                if (args.Length > 1)
+                    short.TryParse(args[1], out speed);
+                user.Effect(effect, speed);
+                return Success($"Displayed effect {effect}.");
+            }
+            return Fail("The value you specified could not be parsed (byte)");
+        }
+    }
+
+    class SoundCommand : ChatCommand
+    {
+        public new static string Command = "sound";
+        public new static string ArgumentText = "<byte sound>";
+        public new static string HelpText = "Plays the specified sound effect.";
+        public new static bool Privileged = false;
+        public new ChatCommandResult Run(User user, params string[] args)
+        {
+            if (byte.TryParse(args[1], out byte sound))
+            {
+                user.SendSound(sound);
+                return Success($"Played sound {sound}.");
+            }
+
+            return Fail("The value you specified could not be parsed (byte)");
+        }
+    }
+
+    class MusicCommand : ChatCommand
+
+    {
+        public new static string Command = "music";
+        public new static string ArgumentText = "<byte music>";
+        public new static string HelpText = "Plays the specified background music.";
+        public new static bool Privileged = false;
+
+        public new ChatCommandResult Run(User user, params string[] args)
+        {
+            if (byte.TryParse(args[1], out byte track))
+            {
+                user.Map.Music = track;
+                foreach (var mapuser in user.Map.Users.Values)
+                {
+                    mapuser.SendMusic(track);
+                }
+                return Success($"Music track changed to {track} for all users on {user.Map.Name}.");
+            }
+            return Fail("The value you specified could not be parsed (byte)");
+        }
+    }
+
+    class ItemCommand : ChatCommand
+    {
+        public new static string Command = "item";
+        public new static string ArgumentText = "<string itemName> [<uint quantity>]";
+        public new static string HelpText = "Give yourself the specified item, with optional quantity.";
+        public new static bool Privileged = false;
+
+
+        public new ChatCommandResult Run(User user, params string[] args)
+        {
+            if (Game.World.WorldData.TryGetValueByIndex(args[0], out Item template))
+            {
+                var item = Game.World.CreateItem(template.Id);
+                if (int.TryParse(args[1], out int count) && count <= item.MaximumStack)
```

Comment:
Need to do a length check on args[] otherwise, this becomes an IndexException.

Reply:
fixed